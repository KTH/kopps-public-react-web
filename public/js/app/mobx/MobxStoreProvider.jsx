/* eslint no-use-before-define: ["error", "nofunc"] */

import React from 'react'

import { createMobxManager } from './manager'

export default MobxStoreProvider

/**
 * React-component which prepares a MobX-store
 * so that its data can be used with the useStore() hook
 * of any other component embedded into this Provider.
 *
 * @param {object} props
 * @param {function} props.initCallback
 * @param {string} [props.storeId]
 * @param {object} props.children
 *
 */
function MobxStoreProvider(props) {
  const { initCallback, storeId, children } = props

  const manager = createMobxManager(storeId || 'default', initCallback)

  const { Provider: ReactContextProvider, value: mobxStore } = manager.getProviderComponentAndValue()

  return <ReactContextProvider value={mobxStore}>{children}</ReactContextProvider>
}
