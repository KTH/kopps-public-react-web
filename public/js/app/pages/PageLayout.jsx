import React from 'react'
import { Col, Row } from 'reactstrap'
import MainMenu from '../components/MainMenu'

function MainContent({ children }) {
  return <main id="mainContent">{children}</main>
}

function PageLayout({ children }) {
  return (
    <Row style={{ padding: '30px 0' }}>
      <MainMenu />
      <Col>
        <MainContent>{children}</MainContent>
      </Col>
    </Row>
  )
}

export default PageLayout
