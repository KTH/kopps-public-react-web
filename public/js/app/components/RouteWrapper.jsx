/* eslint-disable react/jsx-props-no-spreading */
import React from 'react'
import { Route } from 'react-router-dom'
import PropTypes from 'prop-types'
import { MobxStoreProvider } from '../mobx'

function RouteWrapper({
  createBreadcrumbs,
  component: Component,
  layout: Layout,
  applicationStore,
  createMenuData,
  ...rest
}) {
  const menuData = createMenuData(applicationStore)
  const breadcrumbs = createBreadcrumbs(applicationStore)
  return (
    <MobxStoreProvider initCallback={() => applicationStore}>
      <Route
        {...rest}
        render={() => (
          <Layout breadcrumbs={breadcrumbs} menuData={menuData}>
            <Component />
          </Layout>
        )}
      />
    </MobxStoreProvider>
  )
}

RouteWrapper.propTypes = {
  createBreadcrumbs: PropTypes.func,
  component: PropTypes.oneOfType([PropTypes.node, PropTypes.func, PropTypes.elementType]).isRequired,
  layout: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
  createMenuData: PropTypes.func.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  applicationStore: PropTypes.object.isRequired,
}

RouteWrapper.defaultProps = {
  createBreadcrumbs: () => ({ include: 'student', items: [] }),
}

export default RouteWrapper
