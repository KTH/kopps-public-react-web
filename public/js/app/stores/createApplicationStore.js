/* eslint no-use-before-define: ["error", "nofunc"] */

// @ts-check

// eslint-disable-next-line no-unused-vars
import { observable } from 'mobx'
import commonStore from './commonStore'
import curriculumStore from './curriculumStore'
import studyProgrammeStore from './studyProgrammeStore'
import searchCoursesStore from './searchCoursesStore'
import appendix1Store from './appendix1Store'
import appendix2Store from './appendix2Store'
import literatureStore from './literatureStore'

export default createApplicationStore

const store = {
  /**
   * @property {string} message
   */
  message: 'Hallo',
  /**
   * @method
   * @param {string} text
   */
  setMessage,
  /**
   * @property {string} lastAdmissionTerm
   */
  lastAdmissionTerm: '',
  /**
   * @method
   * @param {string} lastAdmissionTerm
   */
  setLastAdmissionTerm,
  /**
   * @method
   * @param {map<string, {}>} programmes
   */
  setProgrammes,

  /**
   * @property {map<string, {}>} programmes
   */
  programmes: [],

  /**
   * @method
   * @param {[]} currentSchoolsWithDepartments
   */
  setCurrentSchoolsWithDepartments,

  /**
   * @property {[]} currentSchoolsWithDepartments
   */
  currentSchoolsWithDepartments: [],

  /**
   * @method
   * @param {[]} deprecatedSchoolsWithDepartments
   */
  setDeprecatedSchoolsWithDepartments,

  /**
   * @property {[]} deprecatedSchoolsWithDepartments
   */
  deprecatedSchoolsWithDepartments: [],
  /**
   * @property {string} departmentName
   */
  departmentName: '',
  /**
   * @method
   * @param {string} departmentName
   */
  setDepartmentName,

  /**
   * @property {[]} departmentCourses
   */
  departmentCourses: [],
  /**
   * @method
   * @param {[]} departmentCourses
   */
  setDepartmentCourses,
  /**
   * @property {string} programmeName
   */
  programmeName: '',
  /**
   * @method
   * @param {string} programmeName
   */
  setProgrammeName,
  /**
   * @property {string} programmeCode
   */
  programmeCode: '',
  /**
   * @method
   * @param {string} programmeCode
   */
  setProgrammeCode,
  /**
   * @property {[]} programmeTerms
   */
  programmeTerms: [],
  /**
   * @method
   * @param {[]} programmeTerms
   */
  setProgrammeTerms,
  /**
   * @property {number} lengthInStudyYears
   */
  lengthInStudyYears: 0,
  /**
   * @method
   * @param {number} lengthInStudyYears
   */
  setLengthInStudyYears,
  /**
   * @property {string} thisHostBaseUrl
   */
  thisHostBaseUrl: null,
}

function createApplicationStore(storeId) {
  switch (storeId) {
    case 'curriculum':
      return {
        ...commonStore,
        ...curriculumStore,
      }
    case 'searchCourses':
      return {
        ...commonStore,
        ...searchCoursesStore,
      }
    case 'objectives':
    case 'extent':
    case 'eligibility':
    case 'implementation':
      return {
        ...commonStore,
        ...studyProgrammeStore,
      }
    case 'appendix1':
      return {
        ...commonStore,
        ...studyProgrammeStore,
        ...appendix1Store,
      }
    case 'appendix2':
      return {
        ...commonStore,
        ...studyProgrammeStore,
        ...appendix2Store,
      }
    case 'literatureList':
      return {
        ...commonStore,
        ...literatureStore,
      }
    default:
      return {
        ...commonStore,
        ...store,
      }
  }
}

function setMessage(text = 'Happy coding!! :)') {
  this.message = text
}

function setProgrammes(programmes) {
  this.programmes = [...programmes]
}

function setCurrentSchoolsWithDepartments(currentSchoolsWithDepartments) {
  this.currentSchoolsWithDepartments = currentSchoolsWithDepartments
}

function setDeprecatedSchoolsWithDepartments(deprecatedSchoolsWithDepartments) {
  this.deprecatedSchoolsWithDepartments = deprecatedSchoolsWithDepartments
}

function setDepartmentName(departmentName) {
  this.departmentName = departmentName
}

function setDepartmentCourses(departmentCourses) {
  this.departmentCourses = departmentCourses
}

function setLastAdmissionTerm(lastAdmissionTerm) {
  this.lastAdmissionTerm = lastAdmissionTerm
}

function setProgrammeName(programmeName) {
  this.programmeName = programmeName
}

function setProgrammeCode(programmeCode) {
  this.programmeCode = programmeCode
}

function setProgrammeTerms(programmeTerms) {
  this.programmeTerms = programmeTerms
}

function setLengthInStudyYears(lengthInStudyYears) {
  this.lengthInStudyYears = lengthInStudyYears
}
