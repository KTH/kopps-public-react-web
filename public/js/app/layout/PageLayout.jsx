import React, { useEffect } from 'react'
import ReactDOM from 'react-dom'
import { Col, Row } from 'reactstrap'
import { Breadcrumbs } from '@kth/kth-reactstrap/dist/components/utbildningsinfo'

import { useStore } from '../mobx'
import MainMenu from '../components/MainMenu'

function renderBreadcrumbsIntoKthHeader(language) {
  const breadcrumbContainer = document.getElementById('breadcrumbs-header')
  if (breadcrumbContainer) {
    ReactDOM.render(<Breadcrumbs include="directory" language={language} />, breadcrumbContainer)
  }
}

function MainContent({ children }) {
  return (
    <Col>
      <main id="mainContent">{children}</main>
    </Col>
  )
}

function PageLayout({ children }) {
  const { language } = useStore()
  useEffect(() => renderBreadcrumbsIntoKthHeader(language))
  return (
    // Container in publicLayout.handlebars – begin
    <Row>
      <MainMenu />
      <MainContent>{children}</MainContent>
    </Row>
    // Container – end
  )
}

export default PageLayout
