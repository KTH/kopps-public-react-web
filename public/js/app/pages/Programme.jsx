import React from 'react'
import PropTypes from 'prop-types'
import { Col, Row } from 'reactstrap'
import { Link, LinkList, PageHeading } from '@kth/kth-reactstrap/dist/components/studinfo'

import Article from '../components/Article'
import FooterContent from '../components/FooterContent'

import { useStore } from '../mobx'
import i18n from '../../../../i18n'
import translate from '../util/translate'
import { programTermLink } from '../util/links'
import { studyYear as calculateStudyYear } from '../util/terms'

// TODO: Move to util
function formatTerm(term) {
  const { language } = useStore()
  const t = translate(i18n, language)
  const [year, semester] = term.split(/([1|2])$/)
  const shortYear = year.slice(-2)
  return `${t('semester')[semester]}${language === 'en' ? ' ' : ''}${shortYear}`
}

function programmeTermLinkText(term) {
  const { language } = useStore()
  const t = translate(i18n, language)
  return `${t('programme_admitted_year')} ${formatTerm(term)}`
}

function ProgrammeTermsLinkListItem({ term }) {
  const { language, browserConfig, programmeCode, lengthInStudyYears } = useStore()
  const studyYear = calculateStudyYear(term, lengthInStudyYears)
  return (
    <Link
      href={programTermLink(browserConfig.proxyPrefixPath.uri, programmeCode, term, `arskurs${studyYear}`, language)}
    >
      {programmeTermLinkText(term)}
    </Link>
  )
}

function ProgrammeTermsLinkList() {
  const { programmeTerms } = useStore()
  return (
    <LinkList>
      {programmeTerms.map(term => (
        <ProgrammeTermsLinkListItem key={term} term={term} />
      ))}
    </LinkList>
  )
}

function Programme() {
  const { language, programmeCode, programmeName } = useStore()
  const t = translate(i18n, language)
  return (
    <>
      <Row>
        <Col>
          <PageHeading subHeading={`${programmeName} (${programmeCode})`}>{t('programme_study_years')}</PageHeading>
        </Col>
      </Row>
      <Row>
        <Col>
          <Article>
            <p>{t('programme_study_years_explanation')}</p>
            <ProgrammeTermsLinkList />
          </Article>
        </Col>
      </Row>
      <Row>
        <Col>
          <FooterContent />
        </Col>
      </Row>
    </>
  )
}

ProgrammeTermsLinkListItem.propTypes = {
  term: PropTypes.string.isRequired,
}

export default Programme
