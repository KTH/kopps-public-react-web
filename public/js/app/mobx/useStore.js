/* eslint no-use-before-define: ["error", "nofunc"] */

import createApplicationStore from '../stores/createApplicationStore'
import { getMobxManager } from './manager'

export default useStore

/**
 * Hook which can be used by any React component
 * within the related MobxStoreProvider.
 *
 * @param {string} [storeId]
 *
 *      Current state of application store
 */
function useStore(storeId = 'default') {
  const manager = getMobxManager(storeId)

  const isTestEnvironmentWithoutProviderComponent = manager == null
  if (isTestEnvironmentWithoutProviderComponent) {
    const dummyStoreForTesting = createApplicationStore()
    return dummyStoreForTesting
  }

  return manager.getContextHook()
}
