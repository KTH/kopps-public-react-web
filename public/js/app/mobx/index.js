/* eslint no-use-before-define: ["error", "nofunc"] */

import MobxStoreProvider from './MobxStoreProvider'
import useStore from './useStore'
import { compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument } from './compress'
// import { compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument } from './compressWithPako'

export { MobxStoreProvider, useStore, compressStoreIntoJavascriptCode, uncompressStoreInPlaceFromDocument }
