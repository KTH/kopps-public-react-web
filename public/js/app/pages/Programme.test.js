/* eslint-disable react/prop-types */

import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { axe, toHaveNoViolations } from 'jest-axe'
import { StaticRouter } from 'react-router'
import { MobxStoreProvider } from '../mobx'

import Programme from './Programme'
import RouteWrapper from '../components/RouteWrapper'
import PageLayout from '../layout/PageLayout'

import createApplicationStore from '../stores/createApplicationStore'
import getProgrammeMenuData from '../config/programmeMenuData'

import commonSettings from '../config/mocks/mockCommonSettings'

expect.extend(toHaveNoViolations)

const mockDate = new Date('2021-03-23 16:00')

const applicationStore = createApplicationStore()

const testProgrammeCode = 'ARKIT'
const testProgrammeName = {
  sv: 'Arkitektutbildning',
  en: 'Degree Programme in Architecture',
}
const testLengthInStudyYears = 5
const testProgrammeTerms = ['20192', '20211']

const WrapperProgramme = ({ lang }) => {
  applicationStore.setLanguage(lang)
  applicationStore.setBrowserConfig(commonSettings)
  const updatedApplicationStore = {
    ...applicationStore,
  }
  return (
    <StaticRouter>
      <RouteWrapper
        exact
        path="/student/kurser/program/CLGYM"
        createBreadcrumbs={() => ({ include: 'directory' })}
        component={Programme}
        layout={PageLayout}
        applicationStore={updatedApplicationStore}
        createMenuData={store => ({ selectedId: 'studyYears', ...getProgrammeMenuData(store) })}
      />
    </StaticRouter>
  )
}

const ProgrammeWithLayout = ({ lang, lastAdmissionTerm }) => {
  applicationStore.setLanguage(lang)
  applicationStore.setBrowserConfig(commonSettings)
  applicationStore.setProgrammeCode(testProgrammeCode)
  applicationStore.setProgrammeName(testProgrammeName[lang])
  applicationStore.setLengthInStudyYears(testLengthInStudyYears)
  applicationStore.setProgrammeTerms(testProgrammeTerms)
  applicationStore.setLastAdmissionTerm(lastAdmissionTerm) // to test older programs
  const programmeMenuData = getProgrammeMenuData(applicationStore)

  const updatedApplicationStore = {
    ...applicationStore,
  }
  return (
    <StaticRouter>
      <MobxStoreProvider initCallback={() => updatedApplicationStore}>
        <PageLayout
          breadcrumbs={{ include: 'directory' }}
          menuData={{ selectedId: 'studyYears', ...programmeMenuData }}
        >
          <Programme />
        </PageLayout>
      </MobxStoreProvider>
    </StaticRouter>
  )
}

describe('Render component Programme within RouterWrapper', () => {
  test('simple render', done => {
    render(<WrapperProgramme lang="en" />)
    done()
  })
})

describe('Render component Programme within Layout', () => {
  beforeAll(() => {
    jest.spyOn(global, 'Date').mockImplementation(() => mockDate)
  })

  afterAll(() => {
    jest.spyOn(global, 'Date').mockRestore()
  })
  test('verify that page is accessible', async () => {
    const { container } = render(<ProgrammeWithLayout lang="en" />)
    expect(await axe(container)).toHaveNoViolations()
  })
  test('get page header in English', () => {
    render(<ProgrammeWithLayout lang="en" />)
    const h1Header = screen.getByRole('heading', { level: 1 })
    expect(h1Header).toHaveTextContent('Programme Syllabuses')
  })
  test('get page header in Swedish', () => {
    render(<ProgrammeWithLayout lang="sv" />)
    const h1Header = screen.getByRole('heading', { level: 1 })
    expect(h1Header).toHaveTextContent('Utbildningsplaner')
  })
  test('match to snapshot in English', done => {
    const { asFragment } = render(<ProgrammeWithLayout lang="en" />)
    expect(asFragment()).toMatchSnapshot()
    done()
  })
  test('match to snapshot in Swedish', done => {
    const { asFragment } = render(<ProgrammeWithLayout lang="sv" />)
    expect(asFragment()).toMatchSnapshot()
    done()
  })
})

describe('Render component Programme and check its menu, content and links', () => {
  beforeAll(() => {
    jest.spyOn(global, 'Date').mockImplementation(() => mockDate)
  })

  afterAll(() => {
    jest.spyOn(global, 'Date').mockRestore()
  })
  test('check all links on the page in English for a programme which is available for new admission', () => {
    render(<ProgrammeWithLayout lang="en" />)
    const links = screen.getAllByRole('link')
    expect(links.length).toBe(5)
    // Menu links
    expect(links[0]).toHaveTextContent('Course and programme directory')
    expect(links[0].href).toStrictEqual('http://localhost/student/kurser?l=en')
    // Main content links
    expect(links[1]).toHaveTextContent('Programme syllabus for studies starting in Autumn 19')
    expect(links[1].href).toStrictEqual('http://localhost/student/kurser/program/ARKIT/20192/arskurs2?l=en')
    expect(links[2]).toHaveTextContent('Programme syllabus for studies starting in Spring 21')
    expect(links[2].href).toStrictEqual('http://localhost/student/kurser/program/ARKIT/20211/arskurs1?l=en')
    // Footer links
    expect(links[3]).toHaveTextContent('Central study counseling')
    expect(links[3].href).toStrictEqual('https://www.kth.se/en/studies/master/general-study-counselling-1.621634')
    expect(links[4]).toHaveTextContent('kopps@kth.se')
    expect(links[4].href).toStrictEqual('mailto:kopps@kth.se')
  })
  test('check all links on the page in English for a programme without new admission', () => {
    render(<ProgrammeWithLayout lang="en" lastAdmissionTerm="20172" />)
    const links = screen.getAllByRole('link')
    expect(links.length).toBe(6)
    // Menu links
    expect(links[0]).toHaveTextContent('Course and programme directory')
    expect(links[0].href).toStrictEqual('http://localhost/student/kurser?l=en')
    // Main content links
    expect(links[1]).toHaveTextContent('Programme syllabus for studies starting in Autumn 19')
    expect(links[1].href).toStrictEqual('http://localhost/student/kurser/program/ARKIT/20192/arskurs2?l=en')
    expect(links[2]).toHaveTextContent('Programme syllabus for studies starting in Spring 21')
    expect(links[2].href).toStrictEqual('http://localhost/student/kurser/program/ARKIT/20211/arskurs1?l=en')
    expect(links[3]).toHaveTextContent('Study Handbook 00/01 to 07/08')
    expect(links[3].href).toStrictEqual('http://localhost/student/program/shb?l=en')
    // Footer links
    expect(links[4]).toHaveTextContent('Central study counseling')
    expect(links[4].href).toStrictEqual('https://www.kth.se/en/studies/master/general-study-counselling-1.621634')
    expect(links[5]).toHaveTextContent('kopps@kth.se')
    expect(links[5].href).toStrictEqual('mailto:kopps@kth.se')
  })
  test('check all links on the page in Swedish for a programme which is available for new admission', () => {
    render(<ProgrammeWithLayout lang="sv" />)
    const links = screen.getAllByRole('link')
    expect(links.length).toBe(5)
    // Menu links
    expect(links[0]).toHaveTextContent('Kurs- och programkatalogen')
    expect(links[0].href).toStrictEqual('http://localhost/student/kurser')
    // Main content links
    expect(links[1]).toHaveTextContent('Utbildningsplan kull HT19')
    expect(links[1].href).toStrictEqual('http://localhost/student/kurser/program/ARKIT/20192/arskurs2')
    expect(links[2]).toHaveTextContent('Utbildningsplan kull VT21')
    expect(links[2].href).toStrictEqual('http://localhost/student/kurser/program/ARKIT/20211/arskurs1')
    // Footer links
    expect(links[3]).toHaveTextContent('Central studievägledning')
    expect(links[3].href).toStrictEqual('https://www.kth.se/studievagledning')
    expect(links[4]).toHaveTextContent('kopps@kth.se')
    expect(links[4].href).toStrictEqual('mailto:kopps@kth.se')
  })

  test('check all links on the page in Swedish for a programme without new admission', () => {
    render(<ProgrammeWithLayout lang="sv" lastAdmissionTerm="20212" />)
    const links = screen.getAllByRole('link')
    expect(links.length).toBe(6)
    // Menu links
    expect(links[0]).toHaveTextContent('Kurs- och programkatalogen')
    expect(links[0].href).toStrictEqual('http://localhost/student/kurser')
    // Main content links
    expect(links[1]).toHaveTextContent('Utbildningsplan kull HT19')
    expect(links[1].href).toStrictEqual('http://localhost/student/kurser/program/ARKIT/20192/arskurs2')
    expect(links[2]).toHaveTextContent('Utbildningsplan kull VT21')
    expect(links[2].href).toStrictEqual('http://localhost/student/kurser/program/ARKIT/20211/arskurs1')
    expect(links[3]).toHaveTextContent('Studiehandboken 00/01 tom 07/08')
    expect(links[3].href).toStrictEqual('http://localhost/student/program/shb')
    // Footer links
    expect(links[4]).toHaveTextContent('Central studievägledning')
    expect(links[4].href).toStrictEqual('https://www.kth.se/studievagledning')
    expect(links[5]).toHaveTextContent('kopps@kth.se')
    expect(links[5].href).toStrictEqual('mailto:kopps@kth.se')
  })
})
