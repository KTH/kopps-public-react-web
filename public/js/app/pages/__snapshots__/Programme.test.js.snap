// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component Programme within Layout match to snapshot in English 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <nav
      aria-labelledby="kth-local-navigation-title--mobile"
      class="kth-local-navigation--mobile"
    >
      <button
        class="kth-button menu"
        id="kth-local-navigation-title--mobile"
      >
        <span>
          Degree Programme in Architecture
        </span>
      </button>
      <dialog
        class="kth-mobile-menu left"
      >
        <div
          class="kth-mobile-menu__navigation"
        >
          <button
            class="kth-icon-button close"
          >
            <span
              class="kth-visually-hidden"
            >
              Close
            </span>
          </button>
        </div>
        <div
          class="mobile-menu__content"
        >
          <a
            class="kth-button back"
            href="/student/kurser?l=en"
          >
            Course and programme directory
          </a>
          <span>
            Degree Programme in Architecture
          </span>
          <ul>
            <li>
              <a
                aria-current="page"
              >
                Programme Syllabuses
              </a>
            </li>
          </ul>
        </div>
      </dialog>
    </nav>
    <nav
      aria-labelledby="local-navigation-title"
      class="kth-local-navigation col"
      id="mainMenu"
    >
      <a
        class="kth-button back"
        href="/student/kurser?l=en"
      >
        Course and programme directory
      </a>
      <h2
        id="local-navigation-title"
      >
        Degree Programme in Architecture
      </h2>
      <ul>
        <li>
          <a
            aria-current="page"
          >
            Programme Syllabuses
          </a>
        </li>
      </ul>
    </nav>
    <main
      class="col"
      id="mainContent"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <header
            role="presentation"
          >
            <h1
              aria-labelledby="page-heading page-sub-heading"
              id="page-heading"
            >
              Programme Syllabuses
            </h1>
            <p
              aria-hidden="true"
              class="page-sub-heading"
              id="page-sub-heading"
            >
              Degree Programme in Architecture (ARKIT)
            </p>
          </header>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <article
            class="article "
          >
            <p>
              Choose programme syllabus. Each year group within a programme has its own programme syllabus. The programme syllabus contains, among other things, information about which courses are included in the programme. 
            </p>
            <ul
              class="link-list"
            >
              <li>
                <a
                  href="/student/kurser/program/ARKIT/20192/arskurs2?l=en"
                >
                  Programme syllabus for studies starting in Autumn 19
                </a>
              </li>
              <li>
                <a
                  href="/student/kurser/program/ARKIT/20211/arskurs1?l=en"
                >
                  Programme syllabus for studies starting in Spring 21
                </a>
              </li>
            </ul>
          </article>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <footer
            class="border-top mt-4 pt-1"
            id="articleFooter"
          >
            <address>
              Contact for questions regarding education at KTH: 
              <a
                href="https://www.kth.se/en/studies/master/general-study-counselling-1.621634"
              >
                Central study counseling
              </a>
              <br />
              Contact regarding technical matters on this page: 
              <a
                href="mailto:kopps@kth.se"
              >
                kopps@kth.se
              </a>
            </address>
          </footer>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Render component Programme within Layout match to snapshot in Swedish 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <nav
      aria-labelledby="kth-local-navigation-title--mobile"
      class="kth-local-navigation--mobile"
    >
      <button
        class="kth-button menu"
        id="kth-local-navigation-title--mobile"
      >
        <span>
          Arkitektutbildning
        </span>
      </button>
      <dialog
        class="kth-mobile-menu left"
      >
        <div
          class="kth-mobile-menu__navigation"
        >
          <button
            class="kth-icon-button close"
          >
            <span
              class="kth-visually-hidden"
            >
              Close
            </span>
          </button>
        </div>
        <div
          class="mobile-menu__content"
        >
          <a
            class="kth-button back"
            href="/student/kurser"
          >
            Kurs- och programkatalogen
          </a>
          <span>
            Arkitektutbildning
          </span>
          <ul>
            <li>
              <a
                aria-current="page"
              >
                Utbildningsplaner
              </a>
            </li>
          </ul>
        </div>
      </dialog>
    </nav>
    <nav
      aria-labelledby="local-navigation-title"
      class="kth-local-navigation col"
      id="mainMenu"
    >
      <a
        class="kth-button back"
        href="/student/kurser"
      >
        Kurs- och programkatalogen
      </a>
      <h2
        id="local-navigation-title"
      >
        Arkitektutbildning
      </h2>
      <ul>
        <li>
          <a
            aria-current="page"
          >
            Utbildningsplaner
          </a>
        </li>
      </ul>
    </nav>
    <main
      class="col"
      id="mainContent"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <header
            role="presentation"
          >
            <h1
              aria-labelledby="page-heading page-sub-heading"
              id="page-heading"
            >
              Utbildningsplaner
            </h1>
            <p
              aria-hidden="true"
              class="page-sub-heading"
              id="page-sub-heading"
            >
              Arkitektutbildning (ARKIT)
            </p>
          </header>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <article
            class="article "
          >
            <p>
              Välj utbildningsplan. Varje årskull inom ett program har en egen utbildningsplan. I utbildningsplanen finns information om bland annat vilka kurser som ingår i programmet. 
            </p>
            <ul
              class="link-list"
            >
              <li>
                <a
                  href="/student/kurser/program/ARKIT/20192/arskurs2"
                >
                  Utbildningsplan kull HT19
                </a>
              </li>
              <li>
                <a
                  href="/student/kurser/program/ARKIT/20211/arskurs1"
                >
                  Utbildningsplan kull VT21
                </a>
              </li>
            </ul>
          </article>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <footer
            class="border-top mt-4 pt-1"
            id="articleFooter"
          >
            <address>
              Kontakt för frågor om utbildning på KTH: 
              <a
                href="https://www.kth.se/studievagledning"
              >
                Central studievägledning
              </a>
              <br />
              Teknisk kontakt om denna sida: 
              <a
                href="mailto:kopps@kth.se"
              >
                kopps@kth.se
              </a>
            </address>
          </footer>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
